File: Easy/Even number of Digits/even_digits.java
class Solution {
    public int findNumbers(int[] nums) {
        int even_digits = 0;
        
        //Goes through all the numbers in array
        for(int i=0; i < nums.length; i++){
            int num = nums[i];
            int digits = 0;
            //System.out.println("number: "+ num);
            
            //WHILE number is not zero, it will keep dividing it until it reaches zero.
            //adds to the digits variables.
            while( num > 0){
                num/=10;
                digits++;
                //System.out.println("\twhile number:" + num);
            }
            //System.out.println("\tPOST while number:" + num + "\t digits: " + digits);
            
            //If digits is even add to even_digits variable to return.
            if(digits%2 == 0){
                even_digits++;
            }
        }
        
        return even_digits;
    }
}

File: Learning/Arrays/Even number of Digits/even_digits.py
class Solution(object):
    def findNumbers(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        even_digits = 0
        
        for num in nums:
            digits = 0
            while num > 0:
                digits += 1
                num /= 10
                
            
            if digits % 2 == 0:
                even_digits += 1
            
        return even_digits
